### PR TITLE
Display route alternatives

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: src/client/components/SuggestedRoutes/SuggestedRouteCard.js
Comment (akafaneh) [inline src/client/components/SuggestedRoutes/SuggestedRouteCard.js]: why not destructuring  the props here to make it shorter and more readable line?

File: src/client/components/SuggestedRoutes/SuggestedRouteDetails.js
Comment (akafaneh) [inline src/client/components/SuggestedRoutes/SuggestedRouteDetails.js]: why this component is class based? why not a functional component? it doesn't  hold any state of it's own or need life cycle hook

File: src/client/components/SuggestedRoutes/SuggestedRouteSummary.js
Comment (akafaneh) [inline src/client/components/SuggestedRoutes/SuggestedRouteSummary.js]: again this one, why is a class component? a rule of thumb: "don't use it if you don't need it". class based components are way heavier than functional one and if you are not using state or life cycle hooks then you don't need it. (out of scope but good to know, with the new react version we don't use classes no more as react introduced [hooks](https://reactjs.org/docs/hooks-intro.html) to have state and side effects on functional components.)

File: src/client/components/SuggestedRoutes/SuggestedRouteSummary.js
Comment (akafaneh) [inline src/client/components/SuggestedRoutes/SuggestedRouteSummary.js]: this is not wrong, but a bad practice/habit, always try to strip out functions out of the render scope if you can.
```
const onSelectRoutClick = () => {
    this.props.handleClick(this.props.id)
}
  render() {
    const { id, isSelected, handleClick, time, price } = this.props;
    return (
      <div>
        {isSelected ? <h4>SELECTED ROUTE</h4> : null}
        <TitleAndDescribe title="HOURS" description={time} />
        <TitleAndDescribe title="TOTAL COST" description={price} />
        {!isSelected ? (
          <button onClick={onSelectRoutClick}>SELECT ROUTE</button>
```
the issue with the way you did it, is that on every render you create a new whole function, when you strip the arrow function out like above, it get created once on the fist load and it just keep referring to it on each render after that, which is good for performance.

File: src/client/components/SuggestedRoutes/SuggestedRoutesList.js
Comment (akafaneh) [inline src/client/components/SuggestedRoutes/SuggestedRoutesList.js]: why you are spreading the old state here before assigning a new one?

File: src/client/components/SuggestedRoutes/SuggestedRoutesList.js
Comment (akafaneh) [inline src/client/components/SuggestedRoutes/SuggestedRoutesList.js]: it's alway better to use more meaningful names, what is a list here? what is i refers to? it good naming make the code more understandable

File: src/client/components/SuggestedRoutes/SuggestedRouteSummary.js
Comment (akafaneh) [inline src/client/components/SuggestedRoutes/SuggestedRouteSummary.js]: this is not wrong, but a bad practice/habit, always try to strip out functions out of the render scope if you can.
```
onSelectRoutClick = () => {
    this.props.handleClick(this.props.id)
}
  render() {
    const { id, isSelected, handleClick, time, price } = this.props;
    return (
      <div>
        {isSelected ? <h4>SELECTED ROUTE</h4> : null}
        <TitleAndDescribe title="HOURS" description={time} />
        <TitleAndDescribe title="TOTAL COST" description={price} />
        {!isSelected ? (
          <button onClick={this.onSelectRoutClick}>SELECT ROUTE</button>
```
the issue with the way you did it, is that on every render you create a new whole function, when you strip the arrow function out like above, it get created once on the fist load and it just keep referring to it on each render after that, which is good for performance.

File: src/client/components/SuggestedRoutes/SuggestedRouteDetails.js
Comment (akafaneh) [inline src/client/components/SuggestedRoutes/SuggestedRouteDetails.js]: why this component is class based? why not a functional component? it doesn't hold any state of it's own or need life cycle hook

File: src/client/components/SuggestedRoutes/SuggestedRouteCard.js
Comment (akafaneh) [inline src/client/components/SuggestedRoutes/SuggestedRouteCard.js]: why not destructuring the props here to make it shorter and more readable line?

**Review by akafaneh — CHANGES_REQUESTED** [review]
nice work over all, i left some comments you can check.

File: src/client/components/SuggestedRoutes/SuggestedRoutesList.js
Comment (zkwsk) [reply src/client/components/SuggestedRoutes/SuggestedRoutesList.js]: I always recommend doing that. You have to pass a full copy of the state right? To preserve immutability?

I know in this case we only have `selectedId` so it won't make a difference, but in case more variables are added to the state later.

File: src/client/components/SuggestedRoutes/SuggestedRoutesList.js
Comment (zkwsk) [reply src/client/components/SuggestedRoutes/SuggestedRoutesList.js]: You mean "what is a list item here", right? Because I agree that `i` is not an idea variable name.

File: src/client/components/SuggestedRoutes/SuggestedRoutesList.js
Comment (zkwsk) [reply src/client/components/SuggestedRoutes/SuggestedRoutesList.js]: Looks like I picked up this habit from using Redux. You should safely be able to remove line 10.

Comment (akafaneh) [reply src/client/components/SuggestedRoutes/SuggestedRoutesList.js]: but setState does this out of the box :D

File: src/client/components/SuggestedRoutes/SuggestedRouteDetails.js
Comment (zkwsk) [inline src/client/components/SuggestedRoutes/SuggestedRouteDetails.js]: Tip: for fragments if you write your import like this:
```
import React, { Fragment } from "react";
```
You will be able to write:
```
<Fragment>...</Fragment>
```

Personally I think that looks a lot cleaner. There is also the shorthand `<>...</>` syntax, but personally I think that looks like a typo so that hurts my eyes a little :D But this is just a matter of personal preference, so no need to change it :)

File: src/client/components/TitleAndDescribe.js
Comment (zkwsk) [inline src/client/components/TitleAndDescribe.js]: As we discussed on Slack, I really don't like the name `TitleAndDescribe`. What if I choose that all of them should have a unit (meter, dollars, etc.), should it then be called `TitleAndDescribeAndUnits`.

So think about the purpose of the component. It is essentially a component that shows a very small chunk of information with a describing label. The best name I could come up with was `Field`. It is very closely related to the component `DataCard` that @Swapna-Sahu did though. I think at some point the `DataCard` should just be a regular `Card` component wrapping this component. But I need to look a little more into Material UI to decide this finally. For now, I will be happy if you call this one `Field`.

File: src/client/stories/suggestedRoutes/suggestedRouteCard.js
Comment (zkwsk) [inline src/client/stories/suggestedRoutes/suggestedRouteCard.js]: A small thing:

These are the labels you see in Storybook. It's not that important, but to satisfy my OCD it would be nice if you will make the labels a little more consistent:
```suggestion
time={text("Time")}
price={text("Price")}
eta={text("ETA")}
sailDays={text("Sail days")}
maxWaveHs={text("Max wave HS")}
avgSpeed={text("Average speed")}
hireCost={text("Hire cost")}
consumption={text("Consumption")}
```

Comment (zkwsk) [reply src/client/stories/suggestedRoutes/suggestedRouteCard.js]: Note that it is only abbreviations that are capitalized.

File: src/client/stories/suggestedRoutes/suggestedRouteDetails.js
Comment (zkwsk) [inline src/client/stories/suggestedRoutes/suggestedRouteDetails.js]: Again, consistency can be improved. See above.

**Review by zkwsk — CHANGES_REQUESTED** [review]

File: src/client/stories/suggestedRoutes/suggestedRouteCard.js
Comment (ghost) [reply src/client/stories/suggestedRoutes/suggestedRouteCard.js]: @zkwsk I didn't understand what should I do exactly!

File: src/client/components/SuggestedRoutes/SuggestedRoutesList.js
Comment (ghost) [reply src/client/components/SuggestedRoutes/SuggestedRoutesList.js]: @zkwsk what is the final decision!

File: src/client/stories/suggestedRoutes/suggestedRouteDetails.js
Comment (ghost) [reply src/client/stories/suggestedRoutes/suggestedRouteDetails.js]: I didn't understand here as well!

**Comment by ghost** [comment]
@akafaneh Thanks a lot to review my code, I tried my best to fix the issues.

File: src/client/components/SuggestedRoutes/SuggestedRouteSummary.js
Comment (ghost) [reply src/client/components/SuggestedRoutes/SuggestedRouteSummary.js]: I tried what you mentioned here but I got some errors and I think if it not neccecry I would like to leave it like this otherwise I need your help.

File: src/client/components/SuggestedRoutes/SuggestedRoutesList.js
Comment (zkwsk) [reply src/client/components/SuggestedRoutes/SuggestedRoutesList.js]: It's line 11 you can remove.

File: src/client/stories/suggestedRoutes/suggestedRouteDetails.js
Comment (ghost) [reply src/client/stories/suggestedRoutes/suggestedRouteDetails.js]: Done ;)

File: src/client/stories/suggestedRoutes/suggestedRouteCard.js
Comment (ghost) [reply src/client/stories/suggestedRoutes/suggestedRouteCard.js]: Done ;)

File: src/client/components/SuggestedRoutes/SuggestedRouteSummary.js
Comment (ghost) [reply src/client/components/SuggestedRoutes/SuggestedRouteSummary.js]: Done ;)

File: src/client/components/Field.js
Comment (Swapna-Sahu) [inline src/client/components/Field.js]: Are your sure title and description both should be h4

File: src/client/stories/suggestedRoutes/suggestedRouteCard.js
Comment (Swapna-Sahu) [inline src/client/stories/suggestedRoutes/suggestedRouteCard.js]: Price is a text or number?

File: src/client/stories/suggestedRoutes/suggestedRouteCard.js
Comment (Swapna-Sahu) [inline src/client/stories/suggestedRoutes/suggestedRouteCard.js]: Sails days also I think in number

File: src/client/stories/suggestedRoutes/suggestedRouteCard.js
Comment (Swapna-Sahu) [inline src/client/stories/suggestedRoutes/suggestedRouteCard.js]: please check all prop type for all

**Review by Swapna-Sahu — CHANGES_REQUESTED** [review]
I have few comments. Please check .

File: src/client/components/Field.js
Comment (ghost) [reply src/client/components/Field.js]: @Swapna-Sahu 
No, but we can style them later, what is your suggestion?

File: src/client/stories/suggestedRoutes/suggestedRouteCard.js
Comment (ghost) [reply src/client/stories/suggestedRoutes/suggestedRouteCard.js]: This is going to be just in Storybook to show the demo, So I don't think it is that much important to change it to number.
but if @zkwsk asks me I can change it!

@Swapna-Sahu thanks for mentioning.

File: src/client/stories/suggestedRoutes/suggestedRouteCard.js
Comment (ghost) [reply src/client/stories/suggestedRoutes/suggestedRouteCard.js]: yes, but it will be in storyBook to show the demo!

File: src/client/stories/suggestedRoutes/suggestedRouteCard.js
Comment (ghost) [reply src/client/stories/suggestedRoutes/suggestedRouteCard.js]: @zkwsk Please let me know if it in necessary to change them?

File: src/client/stories/suggestedRoutes/suggestedRouteCard.js
Comment (zkwsk) [reply src/client/stories/suggestedRoutes/suggestedRouteCard.js]: True, but when we add propTypes checking this will fail in Storybook. So please change it to number.

File: src/client/components/Field.js
Comment (zkwsk) [reply src/client/components/Field.js]: I would suggest:
```suggestion
      <h3>title</h3>
      <p>{description}<p>
```

**Review by zkwsk — CHANGES_REQUESTED** [review]
Hey @Omid-Incarnation!

Since we just added prop-type validation, please set up prop-types for your components before we merge them.

Check Abods PR how to do it here:
https://github.com/HackYourFuture-CPH/ov-class09-fp/pull/101

https://github.com/HackYourFuture-CPH/ov-class09-fp/pull/101

File: src/client/components/Field.js
Comment (ghost) [reply src/client/components/Field.js]: Done :)

File: src/client/stories/suggestedRoutes/suggestedRouteCard.js
Comment (zkwsk) [inline src/client/stories/suggestedRoutes/suggestedRouteCard.js:21]: Great work with the proptypes.

Now you just need to make sure that your stories are passing in all of the prop-types you declared. When I open storybook and open the console I see a lot of errors because here you are for example not passing in an ID or a `isSelected`prop.

Please go over this story as well as the other stories you've added and make sure all props are being fed in correctly :)

**Review by zkwsk — CHANGES_REQUESTED** [review]

File: src/client/stories/suggestedRoutes/suggestedRouteCard.js
Comment (zkwsk) [reply src/client/stories/suggestedRoutes/suggestedRouteCard.js:21]: That probably also means removing the units from the numbers above as we discussed, because you can't pass "$66.808" into a number knob for example.

**Review by zkwsk — CHANGES_REQUESTED** [review]

**Review by zkwsk — DISMISSED** [review]

**Review by Swapna-Sahu — DISMISSED** [review]
Looks good to me but need to exchange the suggested route summary and suggested route card.

**Pull request merged**
